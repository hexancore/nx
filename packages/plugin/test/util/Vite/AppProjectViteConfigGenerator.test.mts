import { AppProjectViteConfigGenerator } from '../../../src/util/Vite/index.mjs';
import { createFakeRuntimeHelper } from '../../helper/functions.js';

describe(AppProjectViteConfigGenerator.constructor.name, () => {
  const runtimeHelper = createFakeRuntimeHelper({
    globs: {
      '**/project.json': ['acme/dep/project.json']
    },
    readFiles: {
      '/root/package.json': '{"name": "@acme-work/source"}',
      '/root/acme-work/apps/acme-some/project.json': '{"implicitDependencies": ["acme-dep"]}'
    }
  });
  const generator = new AppProjectViteConfigGenerator(runtimeHelper);

  test.each([{ mode: 'production' }, { mode: "development" }])('should generate with mode: $mode', ({ mode }) => {
    const current = generator.create({
      workspaceProjectDir: '/root/acme-work/apps/acme-some',
    })({ command: 'build', mode });

    current.plugins = [];

    expect(current).toMatchSnapshot();
  });
});