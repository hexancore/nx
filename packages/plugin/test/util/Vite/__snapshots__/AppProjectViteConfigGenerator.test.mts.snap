// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function should generate with mode: development 1`] = `
{
  "build": {
    "assetsDir": "assets",
    "commonjsOptions": {
      "transformMixedEsModules": true,
    },
    "emptyOutDir": true,
    "minify": true,
    "outDir": "../../dist/apps/acme-some",
    "reportCompressedSize": true,
    "rollupOptions": {
      "external": [
        "vue",
      ],
      "output": {
        "assetFileNames": [Function],
        "chunkFileNames": "assets/js/[name]-[hash]-chunk.js",
        "entryFileNames": "assets/js/[name]-[hash].js",
        "format": "esm",
        "paths": undefined,
      },
    },
  },
  "cacheDir": "../../tmp/vite/app-acme-some",
  "plugins": [],
  "preview": {
    "host": "localhost",
    "port": 20022,
  },
  "resolve": {
    "alias": [
      {
        "customResolver": [Function],
        "find": /\\^@acme-work\\\\/project\\.json-dep-acme\\\\/\\(\\.\\*\\)\\$/,
        "replacement": "/root/libs/libs/project.json/dep/acme/src/$1",
      },
    ],
  },
  "root": "/root/acme-work/apps/acme-some",
  "server": {
    "fs": {
      "strict": false,
    },
    "host": "localhost",
    "port": 20021,
  },
  "test": {
    "coverage": {
      "provider": "v8",
      "reportsDirectory": "../../tmp/coverage/app-acme-some",
    },
    "environment": "jsdom",
    "globals": true,
    "include": [
      "test/**/*.test.{ts,mts,cts,tsx}",
    ],
    "reporters": [
      "default",
    ],
  },
}
`;

exports[`Function should generate with mode: production 1`] = `
{
  "build": {
    "assetsDir": "assets",
    "commonjsOptions": {
      "transformMixedEsModules": true,
    },
    "emptyOutDir": true,
    "minify": true,
    "outDir": "../../dist/apps/acme-some",
    "reportCompressedSize": true,
    "rollupOptions": {
      "external": [
        "vue",
      ],
      "output": {
        "assetFileNames": [Function],
        "chunkFileNames": "assets/js/[name]-[hash]-chunk.js",
        "entryFileNames": "assets/js/[name]-[hash].js",
        "format": "esm",
        "paths": undefined,
      },
    },
  },
  "cacheDir": "../../tmp/vite/app-acme-some",
  "plugins": [],
  "preview": {
    "host": "localhost",
    "port": 20022,
  },
  "resolve": {
    "alias": [
      {
        "customResolver": [Function],
        "find": /\\^@acme-work\\\\/project\\.json-dep-acme\\\\/\\(\\.\\*\\)\\$/,
        "replacement": "/root/dist/libs/libs/project.json/dep/acme/$1",
      },
    ],
  },
  "root": "/root/acme-work/apps/acme-some",
  "server": {
    "fs": {
      "strict": false,
    },
    "host": "localhost",
    "port": 20021,
  },
  "test": {
    "coverage": {
      "provider": "v8",
      "reportsDirectory": "../../tmp/coverage/app-acme-some",
    },
    "environment": "jsdom",
    "globals": true,
    "include": [
      "test/**/*.test.{ts,mts,cts,tsx}",
    ],
    "reporters": [
      "default",
    ],
  },
}
`;
